spec/unit/limit_headers_spec.rb

```diff
@@ -78,6 +78,31 @@ module Transactor
                 expect(result).to eq expected
               end
             end
+
+            context 'and one of them has a remaining of 0' do
+              let(:time) { Time.now.utc }
+
+              let(:report_with_some_remaining) do
+                double(period: Period::Day.new(time),
```

Comment:
From rspec-mock view, I think is is really, really necessary to use object/instance mocks and turn on their verification to prevent them going stale:

https://relishapp.com/rspec/rspec-mocks/v/3-8/docs/verifying-doubles/using-an-object-double
https://relishapp.com/rspec/rspec-mocks/v/3-8/docs/verifying-doubles/using-an-instance-double

If it is too much trouble to set up, ok, but something to bear in mind. When using mocks it is super important to verify the mocked interface.

Reply:
Definitely, something to take into account for the future. We're using object_doubles only in a couple of places.
For now, I'd keep it as it is. This stubs the `UsageReport` class which is not easy to instantiate because of the params it receives and improving this could take a bit of time.

Reply:
`instance_double` should skip the initializer (at least that's my understanding), or you can use object double with `UsageReport.allocate` without calling the initializer. It is really just about the interface.

But sure, 👍 if you want to keep it this way.

Reply:
`instance_double` does not skip the initializer.
I like the `.allocate` solution though, so I'll use that :+1: